### PR TITLE
Fixed Title not Showing up on Browser Tabs

### PR DESCRIPTION
**Pull request by ehsan-karamad**
The original implementation of the new-window and user-agent, which are the skeleton for multi-tab browser are based on [\<webview>.executeScript](https://developer.chrome.com/apps/tags/webview#method-executeScript). I initially replaced executeScript with the [less racy API](https://code.google.com/p/chromium/issues/detail?id=489710), [\<webview>.addContentScripts](https://developer.chrome.com/apps/tags/webview#method-addContentScripts). However, the structure of postMessage-ing was not  changed accordingly and hence, the guest (inside the webview) could not communicate with the embedder.

While at this, I also fixed some minor issues with the close dialog which shows up when exiting the last tab. It can now show focused button and also tab + enter keys work.

### CODE REVIEW

File: samples/webview-samples/multi-tab-browser/config.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // Global configuration variables
 var config = {
-  'homepage': 'http://www.google.com/',
+  'homepage': 'http://www.google.com',
```
Comment (beaufortfrancois) [inline samples/webview-samples/multi-tab-browser/config.js]: Let's make it http**s** then ;)

File: samples/webview-samples/multi-tab-browser/guest_messaging.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ var webviewTitleInjectionComplete = false;
     var title = null;
     var postTitle = (function() {
       return function(e) {
-        title = document.title;
+        title = document.title;
```
Comment (beaufortfrancois) [inline samples/webview-samples/multi-tab-browser/guest_messaging.js]: Can you remove these extra white spaces?

File: samples/webview-samples/multi-tab-browser/tabs.js [hunk 3]
```diff
@@ -190,6 +188,13 @@ var tabs = (function(popupModule, contextMenuModule) {
     this.webview.setAttribute('data-name', this.name);
     this.webviewContainer.setAttribute('data-name', this.name);
     this.webviewContainer.classList.add('webview-container');
+    this.webview.addContentScripts([
+          {
+            'name': 'Messageing',
```
Comment (beaufortfrancois) [inline samples/webview-samples/multi-tab-browser/tabs.js]: s/Messageing/Messaging/

File: samples/webview-samples/multi-tab-browser/tabs.js [hunk 4]
```diff
@@ -268,36 +272,13 @@ var tabs = (function(popupModule, contextMenuModule) {
...
-      // Send a message to the webview so it can get a reference to
-      // the embedder
-      var data = {
-        'type': 'titleRequest',
-        'tabName': this.name
-      };
-      this.webview.contentWindow.postMessage(JSON.stringify(data), '*');
-    }
+  Tab.prototype.doContentLoad = function(e) {    
+    // TODO: add the code which is to be run after 'contentload' here.
```
Comment (beaufortfrancois) [inline samples/webview-samples/multi-tab-browser/tabs.js]: Is this TODO still relevant?

**Comment by beaufortfrancois** [comment]
Thanks for improving this sample!
See my inline comments above.

File: samples/webview-samples/multi-tab-browser/config.js (hunk 1 above)
Comment (ehsan-karamad) [inline samples/webview-samples/multi-tab-browser/config.js]: Done.

File: samples/webview-samples/multi-tab-browser/guest_messaging.js (hunk 2 above)
Comment (ehsan-karamad) [inline samples/webview-samples/multi-tab-browser/guest_messaging.js]: Done. Sorry for that. Leaving trailing space appears to be my signature now ;).

File: samples/webview-samples/multi-tab-browser/tabs.js (hunk 3 above)
Comment (ehsan-karamad) [inline samples/webview-samples/multi-tab-browser/tabs.js]: Done.

File: samples/webview-samples/multi-tab-browser/tabs.js (hunk 4 above)
Comment (ehsan-karamad) [inline samples/webview-samples/multi-tab-browser/tabs.js]: Not anymore. I removed it and so did with the comment since it was somewhat obvious.

**Comment by ehsan-karamad** [comment]
Thanks for the review! and sorry it tooke a while until I addressed the issues.

**Comment by beaufortfrancois** [comment]
LGTM!

**Comment by beaufortfrancois** [comment]
Updated at https://chrome.google.com/webstore/detail/nfcmophndjlljioblddmepjbcfnocnak
Thank you!

**Comment by ehsan-karamad** [comment]
Thank you!

**Pull request merged**